crates/contract/src/tee/tee_state.rs

```diff
@@ -338,30 +338,51 @@ impl TeeState {
             .get(tls_public_key)
             .map(|(node_id, _)| node_id.clone())
     }
-    pub fn is_caller_an_attested_node(&self) -> bool {
+    /// Returns true if the caller has at least one participant entry
+    /// whose TLS key matches an attested node belonging to the caller account.
+    ///
+    /// Handles multiple participants per account and supports legacy mock nodes.
+    pub fn is_caller_an_attested_participant(&self, participants: &Participants) -> bool {
         let signer_pk = env::signer_account_pk();
         let signer_id = env::signer_account_id();
-        //let predecessor = env::predecessor_account_id(); //TODO (#1288)  check what is better predecessor or signer
-
-        // Check if caller matches an attested node
-        let result = self.participants_attestations.values().any(|(node_id, _)| {
-            match &node_id.account_public_key {
-                Some(pk) => pk == &signer_pk,
-                None => {
-                    // TODO (#823) Legacy fallback for mock nodes (no account_public_key) -remove when TEE is enforced
-                    node_id.account_id == signer_id
+
+        // find all participant entries that belong to the caller
+        for (account_id, _pid, participant_info) in participants.participants().iter() {
```

Comment:
It feels like it would be sufficient to use `.iter().find(...)` here to simplify the code a bit and avoid some nesting.